GitUI/CommandsDialogs/FormRemotes.cs

```diff
@@ -703,5 +705,84 @@ private void ShowSeparatePushUrl(bool visible)
                 ? _labelUrlAsFetch.Text
                 : _labelUrlAsFetchPush.Text;
         }
+
+        public class ComboBox : System.Windows.Forms.ComboBox
+        {
+            private AutoCompleteMode _autoCompleteMode = AutoCompleteMode.SuggestAppend;
+            private string _lastTextChangedValue;
+
+            private bool SystemAutoCompleteEnabled => _autoCompleteMode != AutoCompleteMode.None && DropDownStyle != ComboBoxStyle.DropDownList;
+
+            protected override void OnValidating(CancelEventArgs e)
+            {
+                if (SystemAutoCompleteEnabled)
+                {
+                    NotifyAutoComplete();
+                }
+            }
+
+            protected override void OnTextChanged(EventArgs e)
+            {
+                if (SystemAutoCompleteEnabled)
+                {
+                    var text = Text;
+                    if (text == _lastTextChangedValue)
+                    {
+                        return;
+                    }
+
+                    base.OnTextChanged(e);
+                    _lastTextChangedValue = text;
+                }
+                else
+                {
+                    base.OnTextChanged(e);
+                }
+            }
+
+            private int FindStringExactCase(string s)
+            {
+                for (var i = 0; i < Items.Count; i++)
+                {
+                    if (((GitCommands.UserRepositoryHistory.Repository)Items[i]).Path.Equals(s, StringComparison.Ordinal))
+                    {
+                        return i;
+                    }
+                }
+
+                return -1;
+            }
+
+            private void NotifyAutoComplete()
+            {
+                var text = Text;
+                var flag1 = text != _lastTextChangedValue;
+                var flag2 = false;
+                int stringIgnoreCase;
```

Comment:
Better name?

Reply:
Renamed to reflect the equivalent naming in ComboBox reference source